nncf/__init__.py

```diff
@@ -22,28 +22,39 @@
 from nncf.quantization import quantize
 from nncf.quantization import quantize_with_accuracy_control
 
+_LOADED_FRAMEWORKS = {
+    "torch": True,
+    "tensorflow": True,
+    "onnx": True,
+    "openvino": True
+}
+
 try:
     import torch
 except ImportError:
-    torch = None
+    _LOADED_FRAMEWORKS["torch"] = False
 
 try:
     import tensorflow as tf
 except ImportError:
-    tf = None
+    _LOADED_FRAMEWORKS["tensorflow"] = False
 
 try:
     import onnx
 except ImportError:
-    onnx = None
+    _LOADED_FRAMEWORKS["onnx"] = False
 
 try:
     import openvino.runtime as ov_runtime
     import openvino.tools.pot as ov_pot
 except ImportError:
-    ov_runtime = None
-    ov_pot = None
+    _LOADED_FRAMEWORKS["openvino"] = False
+
 
-if torch is None and tf is None and onnx is None and ov_runtime is None and ov_pot is None:
-    import warnings
-    warnings.warn("None of PyTorch, TensorFlow, ONNX, OpenVINO have been found. Please, install one of the frameworks")
+if not any(_LOADED_FRAMEWORKS.values()):
+    nncf_logger.error("Neither PyTorch, TensorFlow, ONNX or OpenVINO Python packages have been found in your Python "
```

Comment:
I support. This is a normal practice.

Reply:
You don't raise exceptions on imports in general if this is something you can control.
It's pretty easy to trigger this too - create a clean venv, install NNCF via `pip install .` and run `python -c "import nncf"`.

Reply:
But what's the point of such an environment if nncf cannot work without any of the supported frameworks (openvino should be the case as well)?

Reply:
One way that throwing an exception on import can go wrong is if you have an integration (like in Optimum) that checks if NNCF is installed and then does some imports. That's currently harmless, you may have NNCF installed in your environment for historic reasons, or because it came with openvino-dev (at some point). That doesn't mean that you are intentionally using it there. We can change the optimum integration - but it is just one example that shows how throwing an exception on import may have side effects.

IMO this is exactly what a warning with error level is meant for. An exception is generally expected the moment things go wrong, not because something is expected to go wrong at some point in the future.

Reply:
Thanks @helena-intel, this makes some sense. Let’s skip it for now. But even having openvino with nncf enables post-training scenarios. Single nncf without frameworks is useless.